### ISSUE TITLE
fuzzer

### ISSUE THREAD

**Issue by dgryski** [gonzus/argus#1]
```
#include <stdint.h>
#include <stddef.h>
#include "argus.h"

int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
    Argus* argus = argus_create();
    argus_clear(argus);
    argus_validate_buffer(argus, (const char *)Data, (int)(Size));
    argus_destroy(argus);
    return 0;
}
```

Build with:
```
$ clang -I. -g -fsanitize=fuzzer,address fuzz.c argus.c buffer.c log.c stack.c
```

Finds:
```
SUMMARY: AddressSanitizer: heap-buffer-overflow argus.c:91 in argus_validate_buffer
```

**Comment by gonzus** [comment]
Fixed with fff77e13c98f40b2a78961e2c6450b57ab9a4bae

Can you recommend a good fuzzer for linux + gcc?

**Comment by dgryski** [comment]
clang + libfuzzer is really the way to go for in-process fuzzing.  The others (like afl) are process-based which will make them *much* slower.
